lib/cext/preprocess.rb

```diff
@@ -49,7 +62,35 @@
                  'websocket_mask.c'    => {:gem   => 'websocket-driver',
                                            :patches => 
                                              [{:match       => /(VALUE .*?)\s+= Qnil;/,
-                                               :replacement => '\1;'}]}}
+                                               :replacement => '\1;'}]},
+                 'http11_parser.c'    => {:gem   => 'puma_http11',
+                                           :patches => 
+                                             [{:match       => /parser->(.*?) = Qnil/,
+                                               :replacement => 'parser->\1 = rb_tr_handle_for_managed(Qnil)'}]},
+                 'puma_http11.c'      => {:gem   => 'puma_http11',
+                                           :patches => 
+                                             [{:match       => /(define.*?,) Qnil }/,
+                                               :replacement => '\1 NULL }'},
+                                               {:match       => /cf->value = (.*?);/,
+                                               :replacement => 'cf->value = rb_tr_handle_for_managed(\1);'},
+                                               {:match       => /return found \? found->value : Qnil;/,
+                                               :replacement => 'return found ? rb_tr_managed_from_handle(found->value) : Qnil;'},
+                                               {:match       => /return cf->value;/,
+                                               :replacement => 'return rb_tr_managed_from_handle(cf->value);'},
+                                               {:match       => /void HttpParser_mark\(puma_parser\* hp\) {.*?}/m,
+                                               :replacement => 'void HttpParser_mark(puma_parser* hp) { }'},
+                                               {:match       => /void HttpParser_free\(void \*data\) {.*?}.*?}/m,
```

Comment:
We could match the end of the function with `^\}` if we assume indentation is correct.